service/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.exception.ErrorCode;
+import com.epam.esm.exception.ServiceException;
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.model.Order;
+import com.epam.esm.model.User;
+import com.epam.esm.model.dto.OrderDto;
+import com.epam.esm.model.dto.serialization.DtoSerializer;
+import com.epam.esm.persistence.repository.GiftCertificateRepository;
+import com.epam.esm.persistence.repository.OrderRepository;
+import com.epam.esm.persistence.repository.UserRepository;
+import com.epam.esm.service.OrderService;
+import com.epam.esm.service.util.PaginationUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of {@link OrderService}.
+ */
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final UserRepository userRepository;
+    private final GiftCertificateRepository certificateRepository;
+    private final DtoSerializer<OrderDto, Order> orderDtoSerializer;
+
+    @Autowired
+    public OrderServiceImpl(OrderRepository orderRepository,
+                            UserRepository userRepository,
+                            GiftCertificateRepository certificateRepository,
+                            @Qualifier("orderDtoSerializer") DtoSerializer<OrderDto, Order> orderDtoSerializer) {
+        this.orderRepository = orderRepository;
+        this.userRepository = userRepository;
+        this.certificateRepository = certificateRepository;
+        this.orderDtoSerializer = orderDtoSerializer;
+    }
+
+    @Override
+    public List<OrderDto> fetchAllOrders(int page, int size) {
+        page = PaginationUtil.correctPageIndex(page, size, orderRepository::countAll);
+        return orderRepository.findAll(page, size).stream()
+                .map(orderDtoSerializer::dtoFromEntity)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public OrderDto fetchOrderById(Long id) {
+        Order order = Optional.ofNullable(orderRepository.findById(id))
+                .orElseThrow(() -> new ServiceException(ErrorCode.ORDER_NOT_FOUND, id));
+        return orderDtoSerializer.dtoFromEntity(order);
+    }
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, Long certificateId, int quantity) {
```

Comment:
*minor defect*
Certificate, Tag services has method addCertificate , addTag
why this method has different name?

Reply:
The wording seemed more appropriate. "Place order" better conveys what this service operation on the resource does. So I went with better readability rather than uniformity with other services contracts.